### PR TITLE
protodoc: single source-of-truth for doc protos.

### PR DESCRIPTION
**Pull request by htuch**
This avoids having to list new docs protos in both docs/build.sh and
api/docs/BUILD. This technical debt cleanup is helpful in v3 proto work
to simplify collecting proto artifacts from a Bazel aspect.

Risk level: Low
Testing: docs/build.sh, visual inspection of docs.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: tools/protodoc/protodoc.py [hunk 1]
```diff
@@ -713,7 +713,14 @@ def Main():
   response = plugin_pb2.CodeGeneratorResponse()
   cprofile_enabled = os.getenv('CPROFILE_ENABLED')
 
-  for proto_file in request.proto_file:
+  for file_to_generate in request.file_to_generate:
```
Comment (mattklein123) [inline tools/protodoc/protodoc.py]: Why is this change needed? Presumably this is so that the output maps to what bazel query expects? Can you add some more comments?

File: docs/build.sh [hunk 2]
```diff
@@ -50,126 +50,19 @@ bazel build ${BAZEL_BUILD_OPTIONS} @envoy_api//docs:protos --aspects \
...
-  /envoy/type/matcher/metadata/envoy/type/matcher/metadata.proto.rst
-  /envoy/type/matcher/value/envoy/type/matcher/value.proto.rst
-  /envoy/type/matcher/number/envoy/type/matcher/number.proto.rst
-  /envoy/type/matcher/regex/envoy/type/matcher/regex.proto.rst
-  /envoy/type/matcher/string/envoy/type/matcher/string.proto.rst
-"
-
-# Dump all the generated RST so they can be added to PROTO_RST easily.
-find -L bazel-bin/external/envoy_api -name "*.proto.rst"
+declare -r DOC_PROTOS=$(bazel query  "deps(@envoy_api//docs:protos)"  | grep "^@envoy_api.*proto$")
```
Comment (mattklein123) [inline docs/build.sh]: nit: I think there are some errant spaces in here.

File: tools/protodoc/protodoc.py (hunk 1 above)
Comment (htuch) [reply tools/protodoc/protodoc.py]: Basically, it has to do with the fact that we're invoked under a Bazel aspect, which is a traversal of a shadow build graph. You only want to generate for the current node, not any of the dependencies, otherwise you generate non-canonical versions of files and have O(n^2) rather than O(n) behavior.
